plans/subplans/install.pp

```diff
@@ -125,7 +132,21 @@
   # either be undef or else the key content to write.
   $r10k_private_key = peadm::file_or_content('r10k_private_key', $r10k_private_key_file, $r10k_private_key_content)
 
-  # Same for license key
+  # Determine whether r10k_known_hosts is required and has been provided.
+  $is_pe_2023_3_or_greater = (versioncmp($version, '2023.3.0') >= 0)
+  if (($is_pe_2023_3_or_greater) and
+      ($r10k_private_key =~ NotUndef) and
+      ($r10k_known_hosts =~ Undef)) {
+    fail_plan("In Puppet Enterprise 2023.3+ r10k 4.0 requires host key verification for the r10k_remote host. When setting \$r10k_private_key, you must also provide \$r10k_known_hosts information in the form of an array of hashes with 'name', 'type' and 'key' information for hostname, key-type and public key. Puppet Enterprise version: ${version}, r10k_known_hosts: ${r10k_known_hosts}")
```

Comment:
Few questions here:
* Do we want to fail the plan?
* Is the information in the error message enough to let people fix the problem?
* Do we want to test r10k_remote for a git@ or ssh: protocol instead of testing for existence of r10k_private_key?

Reply:
I think it would be good to add a check on the r10k_remote like `/(^\w+@)|(^ssh://)/`

Reply:
Removed this.